test/select_tests.cpp

```diff
@@ -805,3 +805,123 @@ TEST(IntervalLiteral) {
+
+  stmt = (SelectStatement*)result.getStatement(1);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::None);
+
+  stmt = (SelectStatement*)result.getStatement(2);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForNoKeyUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::None);
+
+  stmt = (SelectStatement*)result.getStatement(3);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForKeyShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::None);
+}
+
+TEST(LockingClauseWithWaitPolicy) {
+  SelectStatement* stmt;
+  TEST_PARSE_SQL_QUERY(
+      "SELECT * FROM t WHERE a = 10 FOR UPDATE NOWAIT;"
+      "SELECT * FROM t WHERE a = 10 FOR SHARE NOWAIT;"
+      "SELECT * FROM t WHERE a = 10 FOR NO KEY UPDATE NOWAIT;"
+      "SELECT * FROM t WHERE a = 10 FOR KEY SHARE NOWAIT;"
+      "SELECT * FROM t WHERE a = 10 FOR UPDATE SKIP LOCKED;"
+      "SELECT * FROM t WHERE a = 10 FOR SHARE SKIP LOCKED;"
+      "SELECT * FROM t WHERE a = 10 FOR NO KEY UPDATE SKIP LOCKED;"
+      "SELECT * FROM t WHERE a = 10 FOR KEY SHARE SKIP LOCKED;",
+      result, 8);
+
+  stmt = (SelectStatement*)result.getStatement(0);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::NoWait);
+
+  stmt = (SelectStatement*)result.getStatement(1);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::NoWait);
+
+  stmt = (SelectStatement*)result.getStatement(2);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForNoKeyUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::NoWait);
+
+  stmt = (SelectStatement*)result.getStatement(3);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForKeyShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::NoWait);
+
+  stmt = (SelectStatement*)result.getStatement(4);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::SkipLocked);
+
+  stmt = (SelectStatement*)result.getStatement(5);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::SkipLocked);
+
+  stmt = (SelectStatement*)result.getStatement(6);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForNoKeyUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::SkipLocked);
+
+  stmt = (SelectStatement*)result.getStatement(7);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForKeyShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::SkipLocked);
+}
+TEST(LockingClauseWithTableReference) {
```

Comment:
Please also check
- the number of elements in `lockings`
- the number of elements in `tables`